libraries/ESP8266WiFi/src/WiFiClient.h

```diff
@@ -55,7 +55,11 @@ class WiFiClient : public Client, public SList<WiFiClient> {
   uint8_t status();
   virtual int connect(IPAddress ip, uint16_t port) override;
   virtual int connect(const char *host, uint16_t port) override;
+  virtual int connectAsync(const char* host, uint16_t port, bool* waiting = nullptr);
```

Comment:
Same comments about `waiting`.  I don't think `nullptr` makes any sense because the app would never know if it was a failure or just that it needs to wait for the answer.